src/Features/Core/Portable/Diagnostics/DiagnosticService.cs

```diff
@@ -81,15 +77,14 @@ private void RaiseDiagnosticsUpdated(IDiagnosticUpdateSource source, Diagnostics
                 }
 
                 ev.RaiseEvent(handler => handler(source, args));
-            }).CompletesAsyncOperation(eventToken);
+            }, CancellationToken.None);
```

Comment:
well that's highly terrifying :)

Reply:
(esp. as i just got a bug talking about how IDE isn't cancelling fast enough in some cases ;-)).  can we file issue on htis?

Reply:
https://github.com/dotnet/roslyn/issues/42679